gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -410,6 +437,113 @@ ChannelCredentials createMtlsChannelCredentials() throws IOException, GeneralSec
     return null;
   }
 
+  /**
+   * This method creates {@link TlsChannelCredentials} to be used by the client to establish an mTLS
+   * connection to S2A. Returns null if any of {@param trustBundle}, {@param privateKey} or {@param
+   * certChain} are missing.
+   *
+   * @param trustBundle the trust bundle to be used to establish the client -> S2A mTLS connection
+   * @param privateKey the client's private key to be used to establish the client -> S2A mtls
+   *     connection
+   * @param certChain the client's cert chain to be used to establish the client -> S2A mtls
+   *     connection
+   * @return {@link ChannelCredentials} to use to create an mtls connection between client and S2A
+   * @throws IOException on error
+   */
+  @VisibleForTesting
+  ChannelCredentials createMtlsToS2AChannelCredentials(
+      InputStream trustBundle, InputStream privateKey, InputStream certChain) throws IOException {
+    if (trustBundle == null || privateKey == null || certChain == null) {
+      return null;
+    }
+    return TlsChannelCredentials.newBuilder()
+        .keyManager(privateKey, certChain)
```

Comment:
`TlsChannelCredentials` also supports passing [File as parameters](https://github.com/grpc/grpc-java/blob/master/api/src/main/java/io/grpc/TlsChannelCredentials.java#L248-L250). This seems a better option since we don't have to worry about closing `InputStreams` or  handling `FileNotFoundException`.

Reply:
This is done in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/a42dcbdf3c2eed8518d14c09dabf737ca99f865b (see discussion in https://github.com/googleapis/sdk-platform-java/pull/3326#discussion_r1838709246)